src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -2888,11 +2894,15 @@ valueReferenceToRawSql sourceIdent columnIdentF info =
 
 aliasedEntityColumnIdent :: Ident -> FieldDef -> IdentInfo -> Ident
 aliasedEntityColumnIdent (I baseIdent) field info =
-  I (baseIdent <> "_" <> (builderToText $ fromDBName info $ fieldDB field))
+  I (baseIdent <> "_" <> (removeTicksFromColumnName $ builderToText $ fromDBName info $ fieldDB field))
 
 aliasedColumnName :: Ident -> IdentInfo -> T.Text -> TLB.Builder 
 aliasedColumnName (I baseIdent) info columnName = 
-  useIdent info (I (baseIdent <> "_" <> columnName))
+  useIdent info (I (baseIdent <> "_" <> (removeTicksFromColumnName columnName)))
+
+removeTicksFromColumnName :: T.Text -> T.Text
+removeTicksFromColumnName columnName = 
+    Maybe.fromMaybe columnName $ T.stripPrefix "`" =<< T.stripSuffix "`" columnName
```

Comment:
The ticks are a MySQL thing. Are we sure that this isn't potentially a problem in Postgres too?

It may be more resilient to get the quote character from the SqlBackend if possible

Reply:
Right so this is probably better solved by changing `Ident` to be `Ident DBName` and just deferring the call to `fromDBName` until the last possible moment. I will probably back out this specific part of the PR since everything should still work with out this and see about changing `Ident`